src/models/Course.js

```diff
@@ -9,12 +9,7 @@ const COURSE_FORMATS = [STRICTLY_E_LEARNING, BLENDED];
 const CourseSchema = mongoose.Schema({
   misc: { type: String },
   subProgram: { type: mongoose.Schema.Types.ObjectId, ref: 'SubProgram', required: true },
-  companies: {
-    type: [mongoose.Schema.Types.ObjectId],
-    default: undefined,
-    ref: 'Company',
-    required() { return this.type === INTRA; },
-  },
+  companies: { type: [mongoose.Schema.Types.ObjectId], ref: 'Company', required() { return this.type === INTRA; } },
```

Comment:
Dans le script de transition, les formations sans stagiaire ont tout de meme le champs companies definie a []. Est-ce qu'il ne faudrait pas faire de meme ici ?

Reply:
Ha pardon ! Je pensais avoir tester mais enfait je n'etais pas sur la bonne branche ^^'
J'ai l'impression que le required ne fonctionne pas 🤔 J'ai commenté le champs "companies" dans un des seeds et je n'ai pas eu d'erreur (la ou si je commente subProgram par exemple ca renvoie bien une erreur)

Reply:
Ah mias oui tu as raison, c'est pcq j'ai retiré la valeur par defaut pour toutes les formations, donc pour les formatiions intra le champ est aussi defini et initialisé a []. Je pense qu'il faut bloquer cette possibilité, que penses-tu d'ajouter (en plus du required) une valeur par defaut comme suit: `default() { return this.type === INTRA ? undefined : []; },` ?

Reply:
On peut peut etre tester leur solution ici https://stackoverflow.com/questions/36860342/mongoose-make-array-required (on est en mongoose 6 donc plutot la premiere solution si ca fonctionne toujours). 
Je me dis que si on trouve un moyen de bloquer c'est mieux parce que sinon ca veut dire qu'on peut techniquement se retrouver avec des formations INTRA sans companies :/

Reply:
Je n'arrive pas la faire fonctionner. J'ai meme essayé ca: https://mongoosejs.com/docs/api/schemaarray.html#schemaarray_SchemaArray-checkRequired
mais si je passe une formatiion INTRA avec `companies: []` ca renvoie pas d'erreur...

Reply:
Enn mixant le lien que t'as envoyé et la doc mongoose: https://mongoosejs.com/docs/validation.html#custom-validators j'ai faiit une validation custom. Qu'en penses-tu ?

Reply:
Trop cool la validation ! :) 
Qu'est-ce qui a motive l'ajout d'un message custom ?

Reply:
Je n'ai pas de raison ^^ Sans msg personnalisé, l'erreur est suffisament claire: " companies: Validator failed for path `companies` with value ``" , je peux la retirer.